src/component/Autocomplete.tsx

```diff
@@ -0,0 +1,101 @@
+import { useCallback, useMemo, useState } from 'react';
+import { peopleFromServer } from '../data/people';
+import { Person } from '../types/Person';
+import debounce from 'lodash.debounce';
+
+type Props = {
+  onSelected: (person: Person | null) => void;
+  delay?: number;
+};
+
+export const AutoComplete: React.FC<Props> = ({ onSelected, delay }) => {
+  const [input, setInput] = useState('');
+  const [showAll, setShowAll] = useState(false);
+
+  const query = useCallback(() => {
```

Comment:
Not sure that there is a need to memoise this function. Have you done it purposely or just in case? :)